views/update/update-manage-trusts-individuals-or-entities-involved.html

```diff
@@ -11,16 +11,295 @@
     <div class="govuk-grid-column-two-thirds">
 
       {% include "includes/list/errors.html" %}
 
+      <span class="govuk-caption-l">{{ pageData.trustData.trustName }}</span>
       <h1 class="govuk-heading-xl">{{ title }}</h1>
+      {% set reviewedBoItems = [] %}
+      {% set addedBoItems = [] %}
+      {% for bo in pageData.boInTrust %}
+        {% set type = 'Beneficial Owner' %}
+        {% if bo.ch_reference %}
+          {% set reviewedBoItems = (
+            reviewedBoItems.push({
+              key: {
+                text: type
+              },
+              value: {
+                text: bo.name
+              }
+            }),
+            reviewedBoItems
+          ) %}
+        {% else %}
+          {% set addedBoItems = (
+            addedBoItems.push({
+              key: {
+                text: type
+              },
+              value: {
+                text: bo.name
+              }
+            }),
+            addedBoItems
+          ) %}
+        {% endif %}
+      {% endfor %}
+
+      {% set reviewedFormerTrustees = [] %}
+      {% set addedFormerTrustees = [] %}
+      {% for trustee in pageData.formerTrusteeData %}
+        {% set type = pageData.trusteeTypeTitle[pageData.trusteeType.HISTORICAL] %}
+        {% if trustee.forename and trustee.surname%}
+          {% set name = trustee.forename + " " + trustee.surname %}
+          {% set formerTrusteeChangeName = OE_CONFIGS.INDIVIDUAL_NAME %}
+        {% else %}
+          {% set name = trustee.corporate_name %}
+          {% set formerTrusteeChangeName = OE_CONFIGS.CORPORATE_NAME %}
+        {% endif %}
+        {% if trustee.ch_references %}
+          {% set reviewedFormerTrustees = (
+            reviewedFormerTrustees.push({
+              key: {
+                text: type  
+              },
+              value: {
+                text: name
+              },
+              actions: {
+                items: [ CREATE_CHANGE_LINK(
+                  OE_CONFIGS.UPDATE_MANAGE_TRUSTS_TELL_US_ABOUT_THE_FORMER_BO_URL + "/" + trustee.id + formerTrusteeChangeName,
+                  "Former Trustee" + trustee.name,
+                  "change-former-trustee-button"
+              ) ]
+              }
+            }),
+            reviewedFormerTrustees
+          ) %}
+        {% else %}
+          {% set addedFormerTrustees = (
+            addedFormerTrustees.push({
+              key: {
+                text: type  
+              },
+              value: {
+                text: name
+              },
+              actions: {
+                items: [ CREATE_CHANGE_LINK(
+                  OE_CONFIGS.UPDATE_MANAGE_TRUSTS_TELL_US_ABOUT_THE_FORMER_BO_URL + "/" + trustee.id + formerTrusteeChangeName,
+                  "Former Trustee" + trustee.name,
+                  "change-former-trustee-button"
+              ) ]
+              }
+            }),
+            addedFormerTrustees
+          ) %}
+        {% endif %}
+      {% endfor %}
```

Comment:
Just a suggestion on simplifying this markup a little, as it appears the row definition added to the different lists is identical.

Also a couple of things that were copied across from the registration page but i think are incorrect (though do correct me if I'm wrong):
- the trustee's name, and # location for the change link, were based off of what names were present in the trustee, whereas there's a `corporate_indicator` property for determining that
- the second parameter to CREATE_CHANGE_LINK (`"Former Trustee " + name`) was using `trustee.name`. Pretty sure this is a 'bug' with registration

Could do the same with the two lists for individuals and legal entities too.

```suggestion
      {% set reviewedFormerTrustees = [] %}
      {% set addedFormerTrustees = [] %}
      {% for trustee in pageData.formerTrusteeData %}
        {% set type = pageData.trusteeTypeTitle[pageData.trusteeType.HISTORICAL] %}
        
        {% set isCorporate = trustee.corporate_indicator === 1 %}
        {% set name = trustee.corporate_name if isCorporate else trustee.forename + " " + trustee.surname %}
        {% set formerTrusteeChangeName = OE_CONFIGS.CORPORATE_NAME if isCorporate else OE_CONFIGS.INDIVIDUAL_NAME %}
        
        {% set row = {
          key: { text: type },
          value: { text: name },
          actions: {
            items: [CREATE_CHANGE_LINK(
              OE_CONFIGS.UPDATE_MANAGE_TRUSTS_TELL_US_ABOUT_THE_FORMER_BO_URL + "/" + trustee.id + formerTrusteeChangeName,
              "Former Trustee " + name,
              "change-former-trustee-button"
            )]
          }
        } %}

        {% if trustee.ch_references %}
          {% set reviewedFormerTrustees = (reviewedFormerTrustees.push(row), reviewedFormerTrustees) %}
        {% else %}
          {% set addedFormerTrustees = (addedFormerTrustees.push(row), addedFormerTrustees) %}
        {% endif %}
      {% endfor %}
```

Reply:
You're right that does simplify better. Changes are up